src/index.js

```diff
@@ -0,0 +1,274 @@
+'use babel'
+
+/* @flow */
+
+import * as Path from 'path'
+import * as FS from 'fs'
+import {
+  getTempDirectory,
+  writeFile,
+  unlinkFile,
+  fileExists,
+  assign,
+  validateExec,
+  exec as execTarget,
+  validateEditor,
+  validateFind
+} from './helpers'
+import type { TextEditor, Range } from 'atom'
+import type { TempFiles, ExecOptions, ExecResult } from './types'
+
+let NamedRegexp = null
+export const FindCache = new Map()
+
+export function exec(
+  command: string,
+  args: Array<string> = [],
+  options: ExecOptions = {}
+): Promise<ExecResult> {
+  validateExec(command, args, options)
+  return execTarget(command, args, options, false)
+}
+
+export function execNode(
+  command: string,
+  args: Array<string> = [],
+  options: ExecOptions = {}
+): Promise<ExecResult> {
+  validateExec(command, args, options)
+  return execTarget(command, args, options, true)
+}
+
+export function rangeFromLineNumber(textEditor: TextEditor, line: number, column: number): Range {
+  validateEditor(textEditor)
+  let lineNumber = line
+
+  if (!Number.isFinite(lineNumber) || Number.isNaN(lineNumber) || lineNumber < 0) {
+    lineNumber = 0
+  }
+
+  const buffer = textEditor.getBuffer()
+  const lineMax = buffer.getLineCount() - 1
+
+  if (lineNumber > lineMax) {
+    throw new Error(`Line number (${lineNumber}) greater than maximum line (${lineMax})`)
+  }
+
+  let colStart = column
+  if (!Number.isFinite(colStart) || Number.isNaN(colStart) || colStart < 0) {
+    const indentation = buffer.lineForRow(lineNumber).match(/^\s+/)
+    if (indentation && indentation.length) {
+      colStart = indentation[0].length
+    } else {
+      colStart = 0
+    }
+  }
+
+  const lineLength = buffer.lineLengthForRow(lineNumber)
+
+  if (colStart > lineLength) {
+    throw new Error(`Column start (${colStart}) greater than line length (${lineLength})`)
+  }
+
+  return [
+    [lineNumber, colStart],
+    [lineNumber, lineLength]
+  ]
+}
+
+export async function findAsync(directory: string, name: string | Array<string>): Promise<?string> {
+  validateFind(directory, name)
+  const names = [].concat(name)
```

Comment:
I like how the previous code showed what this was accomplishing, but this is probably faster so :+1:.